samples/SimpleDataSource/HelloWorldCustomDataProcessor.cs

```diff
@@ -0,0 +1,159 @@
+﻿using Microsoft.Performance.SDK;
+using Microsoft.Performance.SDK.Extensibility;
+using Microsoft.Performance.SDK.Processing;
+using SampleCustomDataSource.Tables;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SampleCustomDataSource
+{
+    //
+    // This is a sample Custom Data Processor that processes simple text files.
+    //
+    // Custom Data Processors are created in Custom Data Source and are used to actually process the files.
+    // An instance of the Processor is created for each file you open whereas only one instance of the Custom Data Source is ever created.
+    //
+    // Data processor is responsible for instantiating the proper tables based on what the user has decided to enable.
+    //
+
+    //
+    // Derive the CustomDataProcessorBase abstract class.
+    //
+
+    public sealed class HelloWorldCustomDataProcessor
+        : CustomDataProcessorBase
+    {
+        private readonly string[] filePaths;
+        private IReadOnlyDictionary<string, IReadOnlyList<Tuple<Timestamp, string>>> fileContent;
+        private DataSourceInfo dataSourceInfo;
+
+        public HelloWorldCustomDataProcessor(
+           string[] filePaths,
+           ProcessorOptions options,
+           IApplicationEnvironment applicationEnvironment,
+           IProcessorEnvironment processorEnvironment,
+           IReadOnlyDictionary<TableDescriptor, Action<ITableBuilder, IDataExtensionRetrieval>> allTablesMapping,
+           IEnumerable<TableDescriptor> metadataTables)
+            : base(options, applicationEnvironment, processorEnvironment, allTablesMapping, metadataTables)
+        {
+            //
+            // Assign the files array to a readonly backing field.
+            //
+
+            this.filePaths = filePaths;
+        }
+
+        public override DataSourceInfo GetDataSourceInfo()
+        {
+            //
+            // The DataSourceInfo is used to tell the SDK the time range of the data(if applicable) and any other relevant data for rendering / synchronizing.
+            // In our case, we have nothing, so just return new DataSourceInfo(0, long.MaxValue) to tell the SDK that it encompasses all time.
```

Comment:
The place where `this.dataSourceInfo` is assigned is not assigning it to `new DataSourceInfo(0, long.MaxValue)`. Is this comment accurate?

Reply:
old comment that needs to be deleted